ios/RNVitals.m

```diff
@@ -0,0 +1,120 @@
+
+#import "RNVitals.h"
+#import <mach/mach.h>
+#import <mach/mach_host.h>
+
+// Used to send events to JS
+#if __has_include(<React/RCTBridge.h>)
+#import <React/RCTBridge.h>
+#import <React/RCTEventDispatcher.h>
+#elif __has_include("RCTBridge.h")
+#import "RCTBridge.h"
+#import "RCTEventDispatcher.h"
+#else
+#import "React/RCTBridge.h"
+#import "React/RCTEventDispatcher.h"
+#endif
+
+@implementation RNVitals
+
+@synthesize bridge = _bridge;
+
+static NSString * const LOW_MEMORY = @"LOW_MEMORY";
+
+- (instancetype)init
+{
+    if ((self = [super init])) {
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(applicationDidReceiveMemoryWarning:) name:UIApplicationDidReceiveMemoryWarningNotification object:nil];
+    }
+
+    return self;
+
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
++ (NSDictionary *)getMemoryInfo
+{
+    struct mach_task_basic_info info;
+    mach_msg_type_number_t size = sizeof(info);
+    kern_return_t kerr = task_info(mach_task_self(),
+                                   MACH_TASK_BASIC_INFO,
+                                   (task_info_t)&info,
+                                   &size);
+
+    if (kerr != KERN_SUCCESS) {
+        return @{@"failed": @"failure"};
```

Comment:
This feels weird to make a psuedo data model just say it failed -- normally we'd raise an error or return `nil`. Return `nil` here